### PR TITLE
bindings/capi: Check invalid parameters in capi.

### PR DESCRIPTION
**Pull request by mihashco**
Added checks to capi bindings to avoid crash when API is used with invalid parameters.

### CODE REVIEW

File: src/bindings/capi/tvgCapi.cpp [hunk 1]
```diff
@@ -83,55 +83,64 @@ TVG_EXPORT Tvg_Canvas* tvg_swcanvas_create()
...
+    if (!canvas) return TVG_RESULT_INVALID_ARGUMENT;
     delete(reinterpret_cast<SwCanvas*>(canvas));
     return TVG_RESULT_SUCCESS;
 }
 
 
 TVG_EXPORT Tvg_Result tvg_swcanvas_set_target(Tvg_Canvas* canvas, uint32_t* buffer, uint32_t stride, uint32_t w, uint32_t h, uint32_t cs)
 {
+    if (!canvas || !buffer) return TVG_RESULT_INVALID_ARGUMENT;
     return (Tvg_Result) reinterpret_cast<SwCanvas*>(canvas)->target(buffer, stride, w, h, static_cast<SwCanvas::Colorspace>(cs));
```
Comment (JSUYA) [inline src/bindings/capi/tvgCapi.cpp:95]: The buffer is being checked inside target function.
If the buffer check is absolutely necessary in this function, I think the color_stop of tvg_gradient_color_stops should be checked for the same reason.
What do you think?

**Review by JSUYA — COMMENTED** [review]
Hi :)
Please check my comment

File: src/bindings/capi/tvgCapi.cpp (hunk 1 above)
Comment (mihashco) [reply src/bindings/capi/tvgCapi.cpp:95]: Buffer check is not necessary here (it not cause crash, so I will prepare fix without buffer check). Thank you

**Review by JSUYA — APPROVED** [review]
LGTM
Thank you for check :+1:

**Pull request merged**